wire-ios-sync-engine/Source/UserSession/SyncStatus.swift

```diff
@@ -118,26 +118,32 @@ public class SyncStatus: NSObject, SyncStatusProtocol, SyncProgress {
     }
 
     public func forceSlowSync() {
-        // Refetch user settings.
-        ZMUser.selfUser(in: managedObjectContext).needsPropertiesUpdate = true
-        // Reset the status.
-        currentSyncPhase = SyncPhase.fetchingLastUpdateEventID
-        RequestAvailableNotification.notifyNewRequestsAvailable(nil)
-        log("slow sync")
-        syncStateDelegate?.didStartSlowSync()
+        managedObjectContext.perform { [weak self] in
```

Comment:
Previously, these code paths were always executed on the sync context (which is what `managedObjectContext` is). Now these methods may be called from async methods, so we need to ensure we switch to the right queue otherwise we'll get some core data errors.